pdns/dnsdistdist/dnsdist-configuration-yaml.cc

```diff
@@ -0,0 +1,1593 @@
+    boost::algorithm::to_lower(frontend->d_provider);
+    frontend->d_proxyProtocolOutsideTLS = bind.tls.proxy_protocol_outside_tls;
+    frontend->d_tlsConfig = std::move(tlsConfig);
+    state.tlsFrontend = std::move(frontend);
+  }
+  else if (protocol == "doq") {
+    auto frontend = std::make_shared<DOQFrontend>();
+    frontend->d_local = ComboAddress(std::string(bind.listen_address), 853);
+    frontend->d_quicheParams.d_tlsConfig = std::move(tlsConfig);
+    frontend->d_quicheParams.d_maxInFlight = bind.doq.max_concurrent_queries_per_connection;
+    frontend->d_quicheParams.d_idleTimeout = bind.quic.idle_timeout;
+    frontend->d_quicheParams.d_keyLogFile = std::string(bind.tls.key_log_file);
+    if (dnsdist::doq::s_available_cc_algorithms.count(std::string(bind.quic.congestion_control_algorithm)) > 0) {
+      frontend->d_quicheParams.d_ccAlgo = std::string(bind.quic.congestion_control_algorithm);
+    }
+    frontend->d_internalPipeBufferSize = bind.quic.internal_pipe_buffer_size;
+    state.doqFrontend = std::move(frontend);
+  }
+  else if (protocol == "doh3") {
+    auto frontend = std::make_shared<DOH3Frontend>();
+    frontend->d_local = ComboAddress(std::string(bind.listen_address), 443);
+    frontend->d_quicheParams.d_tlsConfig = std::move(tlsConfig);
+    frontend->d_quicheParams.d_idleTimeout = bind.quic.idle_timeout;
+    frontend->d_quicheParams.d_keyLogFile = std::string(bind.tls.key_log_file);
+    if (dnsdist::doq::s_available_cc_algorithms.count(std::string(bind.quic.congestion_control_algorithm)) > 0) {
+      frontend->d_quicheParams.d_ccAlgo = std::string(bind.quic.congestion_control_algorithm);
+    }
+    frontend->d_internalPipeBufferSize = bind.quic.internal_pipe_buffer_size;
+    state.doh3Frontend = std::move(frontend);
+  }
+  else if (protocol == "doh") {
+    auto frontend = std::make_shared<DOHFrontend>();
+    frontend->d_tlsContext.d_provider = std::string(bind.tls.provider);
+    boost::algorithm::to_lower(frontend->d_tlsContext.d_provider);
+    frontend->d_library = std::string(bind.doh.provider);
+    if (frontend->d_library == "h2o") {
+#ifdef HAVE_LIBH2OEVLOOP
+      frontend = std::make_shared<H2ODOHFrontend>();
+      // we _really_ need to set it again, as we just replaced the generic frontend by a new one
+      frontend->d_library = "h2o";
+#else /* HAVE_LIBH2OEVLOOP */
+      errlog("DOH bind %s is configured to use libh2o but the library is not available", bind.listen_address);
+      return false;
+#endif /* HAVE_LIBH2OEVLOOP */
+    }
+    else if (frontend->d_library == "nghttp2") {
+#ifndef HAVE_NGHTTP2
+      errlog("DOH bind %s is configured to use nghttp2 but the library is not available", bind.listen_address);
+      return false;
+#endif /* HAVE_NGHTTP2 */
+    }
+    else {
+      errlog("DOH bind %s is configured to use an unknown library ('%s')", bind.listen_address, frontend->d_library);
+      return false;
+    }
+
+    for (const auto& path : bind.doh.paths) {
```

Comment:
While tetsing previously I saw that the default `doh.paths` does not appear to work, and indeed, when not specified `bind.doh.paths` looks empty instead of being filled with an single item '[/dns-query]'. The specification looks like 
```
    - name: "paths"
      type: "Vec<String>"
      default: "/dns-query"
```
which is puzzling, as the default value is not a String sequence but a String.